addons/html_editor/static/src/main/link/link_selection_plugin.js

```diff
@@ -41,105 +41,83 @@ export class LinkSelectionPlugin extends Plugin {
-            singleLinkInSelection.classList.add("o_link_in_selection");
-        }
+    normalize(root) {
+        this.updateFEFFs(root);
+        this.resetLinkInSelection();
     }
 
-    isLinkEligibleForZwnbsp(link) {
-        return (
-            link.isContentEditable &&
-            this.editable.contains(link) &&
-            !this.resources.blacklistLinkZwnbsp?.some((callback) => callback(link))
-        );
+    /**
+     * @param {Element} root
+     */
+    clean(root) {
+        this.removeFEFFs(root);
+        this.clearLinkInSelectionClass(root);
+        this.removeEmptyLinks(root);
     }
 
-    // @todo: unify this and ZWSPlugin's cleanZWS into a single utility
-    removeZWNBSPs(textNode, cursors) {
-        const indexesToRemove = [...textNode.textContent.matchAll(/\uFEFF/g)].map(
-            (match) => match.index
-        );
-        for (const index of indexesToRemove.toReversed()) {
-            textNode.deleteData(index, 1);
-        }
-        cursors.update((cursor) => {
-            if (cursor.node === textNode) {
-                cursor.offset -= indexesToRemove.filter((i) => i < cursor.offset).length;
-            }
-        });
-    }
+    /**
+     * @param {Element} root
+     */
+    updateFEFFs(root) {
+        this.removeFEFFs(root, { exclude: isLegitZwnbsp });
 
-    clearLinkInSelectionClass(root) {
-        // @todo: maybe the querySelectorAll calls should include the root.
-        for (const link of root.querySelectorAll(".o_link_in_selection")) {
-            removeClass(link, "o_link_in_selection");
+        for (const link of selectElements(root, "a")) {
+            this.padLinkWithZws(link);
         }
     }
 
-    clearFEFFs(root) {
-        const cursors = this.shared.preserveSelection();
-        let shouldRestoreCursors = false;
-        for (const node of descendants(root)) {
-            if (node.nodeType === Node.TEXT_NODE && node.textContent.includes("\uFEFF")) {
-                // @todo: isn't rightPos needed as well?
-                // Remove all FEFF within a `prepareUpdate` to make sure to make <br>
-                // nodes visible if needed.
-                const restore = prepareUpdate(...leftPos(node));
-                this.removeZWNBSPs(node, cursors);
-                shouldRestoreCursors = true;
-                restore();
-            }
-        }
-        if (shouldRestoreCursors) {
-            cursors.restore();
-        }
-    }
+    /**
+     * Removes ZWNBSP characters from text nodes within the given root.
+     *
+     * @param {Element} root
+     * @param {Object} [options]
+     * @param {Function} [options.exclude]
+     */
+    removeFEFFs(root, { exclude = () => false } = {}) {
+        const defaultFilter = (node) =>
+            node.nodeType === Node.TEXT_NODE &&
+            node.textContent.includes("\uFEFF") &&
+            node.parentElement.isContentEditable &&
+            !isProtected(node);
```

Comment:
@Goaman In your opinion is this condition (`isProtected`) a particular Odoo concept (and should be moved to the `OdooLinkSelectionPlugin`) or is it a Editor-wide concept that all plugins should know about?

Reply:
@cammarosano This concept, while integral to the editor, could be made into a separate plugin if it doesn't add complexity. Given the work involved, we should maintain it as an editor-wide concept for the time being.